<a href="https://colab.research.google.com/github/codingsohyun/thesis_implementation/blob/main/raw_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive

drive.mount("/content/gdrive")

Mounted at /content/gdrive


In [ ]:
import pandas as pd
import numpy as np

sas = pd.read_sas("./gdrive/MyDrive/심형석교수님/pums.sas7bdat")

In [ ]:
sas

FIPST SECTOR N07_EMPLOYER   RG  TABWGT  EMPLOYMENT_NOISY  \
0        b'23'  b'51'         b'E'  3.0     5.4               9.0   
1        b'23'  b'44'         b'E'  0.0     1.0              74.0   
2        b'23'  b'55'         b'E'  5.0     1.0               2.0   
3        b'23'  b'54'         b'E'  0.0     1.0              34.0   
4        b'23'  b'51'         b'E'  0.0     1.0             120.0   
...        ...    ...          ...  ...     ...               ...   
2165675  b'55'  b'31'         b'E'  0.0     1.0             250.0   
2165676  b'55'  b'48'         b'E'  0.0     1.0              14.0   
2165677  b'25'  b'31'         b'E'  0.0     1.0              74.0   
2165678  b'34'  b'42'         b'E'  0.0     1.0               4.0   
2165679  b'53'  b'42'         b'E'  0.0     1.0             180.0   

         PAYROLL_NOISY  RECEIPTS_NOISY   PCT1  PCT2  ...  NOBUSCRED  \
0                140.0           360.0   74.0  21.0  ...       b'0'   
1               2100.0         20000.0  100.0   0.0  ...       b'0'   
2                 40.0           230.0   27.0  27.0  ...       b'0'   
3               1700.0          6900.0   31.0  31.0  ...       b'0'   
4               3700.0          9100.0   34.0  33.0  ...       b'0'   
...                ...             ...    ...   ...  ...        ...   
2165675         8300.0         46000.0   94.0   2.0  ...       b'0'   
2165676          800.0             0.0   94.0   2.0  ...       b'0'   
2165677         6800.0         16000.0   61.0  21.0  ...       b'0'   
2165678          150.0          1200.0   61.0  21.0  ...       b'0'   
2165679         7500.0         70000.0  100.0   0.0  ...       b'0'   

         NOPERSCRED STARTANOTHER SOLDBUS CEASEOTHER CEASENR CEASENA HUSBWIFE  \
0              b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'2'   
1              b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'0'   
2              b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
3              b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
4              b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
...             ...          ...     ...        ...     ...     ...      ...   
2165675        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
2165676        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
2165677        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
2165678        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
2165679        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   

        FAMILYBUS NUMOWNERS  
0            b'1'      b'3'  
1            b'0'      b'0'  
2            b'1'      b'5'  
3            b'2'      b'6'  
4            b'2'      b'3'  
...           ...       ...  
2165675      b'2'      b'4'  
2165676      b'2'      b'4'  
2165677      b'2'      b'7'  
2165678      b'2'      b'7'  
2165679      b'2'      b'1'  

[2165680 rows x 199 columns]

In [ ]:
sas.shape

(2165680, 199)

In [ ]:
# sas 파일의 데이터프레임
df = pd.DataFrame(sas)

In [ ]:
# df[['PCT1', 'PCT2', 'PCT3', 'PCT4']]
# df[df['PCT1']== 100]
# df[['NUMOWNERS']]

# 'raw' dataframe only includes single-owner firms
raw = df[df['NUMOWNERS']== b'1'].reset_index(drop=True)
raw

FIPST SECTOR N07_EMPLOYER   RG  TABWGT  EMPLOYMENT_NOISY  \
0       b'23'  b'44'         b'E'  0.0   1.000              14.0   
1       b'23'  b'44'         b'E'  0.0   1.000              74.0   
2       b'23'  b'23'         b'E'  0.0   1.000              34.0   
3       b'23'  b'52'         b'N'  7.0  33.611               0.0   
4       b'23'  b'11'         b'E'  0.0   1.000              14.0   
...       ...    ...          ...  ...     ...               ...   
663380  b'13'  b'53'         b'E'  0.0   1.000              14.0   
663381  b'32'  b'53'         b'E'  0.0   1.000               7.0   
663382  b'39'  b'53'         b'E'  0.0   1.000               3.0   
663383  b'48'  b'53'         b'E'  0.0   1.000               4.0   
663384  b'53'  b'42'         b'E'  0.0   1.000             180.0   

        PAYROLL_NOISY  RECEIPTS_NOISY   PCT1  PCT2  ...  NOBUSCRED  \
0              1200.0          8300.0  100.0   0.0  ...       b'0'   
1              1700.0         11000.0  100.0   0.0  ...       b'0'   
2              1600.0          6500.0  100.0   0.0  ...       b'0'   
3                 0.0            10.0  100.0   0.0  ...       b'0'   
4               740.0          3500.0  100.0   0.0  ...       b'0'   
...               ...             ...    ...   ...  ...        ...   
663380          470.0          1500.0  100.0   0.0  ...       b'0'   
663381         3000.0          9600.0  100.0   0.0  ...       b'0'   
663382          300.0           960.0  100.0   0.0  ...       b'0'   
663383          270.0           870.0  100.0   0.0  ...       b'0'   
663384         7500.0         70000.0  100.0   0.0  ...       b'0'   

        NOPERSCRED STARTANOTHER SOLDBUS CEASEOTHER CEASENR CEASENA HUSBWIFE  \
0             b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
1             b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
2             b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
3             b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
4             b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
...            ...          ...     ...        ...     ...     ...      ...   
663380        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
663381        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
663382        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
663383        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   
663384        b'0'         b'0'    b'0'       b'0'    b'2'    b'1'     b'4'   

       FAMILYBUS NUMOWNERS  
0           b'2'      b'1'  
1           b'2'      b'1'  
2           b'2'      b'1'  
3           b'2'      b'1'  
4           b'1'      b'1'  
...          ...       ...  
663380      b'2'      b'1'  
663381      b'2'      b'1'  
663382      b'2'      b'1'  
663383      b'2'      b'1'  
663384      b'2'      b'1'  

[663385 rows x 199 columns]

In [ ]:
# 필요한 데이터만 가진 raw 데이터를 csv 파일로 저장
raw.to_csv('./gdrive/MyDrive/심형석교수님/rraw.csv', index=False)

### raw 데이터 확인하기



In [ ]:
# PCT가 전부 100인지 확인
raw['PCT_100'] = np.where((raw['PCT1'] == 100) | (raw['PCT2'] == 100) | (raw['PCT3'] == 100) | (raw['PCT4'] == 100), 1, 0)
(raw['PCT_100'] == 0).sum()

7861

*   single-owner firm만 추출한 데이터라서 PCT 중 하나만이 전부 100일거라고 생각했는데 그렇지 않은 데이터가 7861개 있음 확인함




In [ ]:
# raw_not_100.shape

In [ ]:
raw_not_100 = raw[raw['PCT_100'] == 0]
raw_not_100[['PCT1', 'PCT2', 'PCT3', 'PCT4', 'EDUC1', 'EDUC2', 'EDUC3', 'EDUC4']].reset_index(drop=True)

PCT1  PCT2  PCT3  PCT4 EDUC1 EDUC2 EDUC3 EDUC4
0     52.0  16.0  16.0  16.0  b'5'  b'3'  b'6'  b'6'
1     71.0  29.0   0.0   0.0  b'5'  b'6'  b'0'  b'0'
2     25.0  25.0  25.0  25.0  b'7'  b'7'  b'7'  b'7'
3     50.0  50.0   0.0   0.0  b'2'  b'6'  b'0'  b'0'
4     64.0  36.0   0.0   0.0  b'3'  b'0'  b'0'  b'0'
...    ...   ...   ...   ...   ...   ...   ...   ...
7856  54.0  42.0   3.0   1.0  b'6'  b'0'  b'0'  b'0'
7857  54.0  42.0   3.0   1.0  b'6'  b'0'  b'0'  b'0'
7858  54.0  42.0   3.0   1.0  b'6'  b'0'  b'0'  b'0'
7859  54.0  42.0   3.0   1.0  b'6'  b'0'  b'0'  b'0'
7860  90.0  10.0   0.0   0.0  b'6'  b'5'  b'0'  b'0'

[7861 rows x 8 columns]